### PR TITLE
Feature - create Azure Table Storage script

### LINKED ISSUE #8: Provide script to create an Azure Storage table
Provide script to create an Azure Storage table

### PR DESCRIPTION
**Pull request by stijnmoreels**
Closes #8

### CODE REVIEW

File: docs/preview/features/powershell/azure-table-storage.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+To have access to the following features, you have to import the module:
+
+```powershell
+PS> Import-Module -Name Arcus.Scripting.TableStorage
+```
+
+## Create new table in Azure Table Storage on Storage Account
+
+(Re)Create a Azure Table Storage based to an Storage Account.
```
Comment (mbraekman) [inline docs/preview/features/powershell/azure-table-storage.md:18]:
```suggestion
(Re)Create a Table Storage within an Azure Storage Account.
```

File: docs/preview/features/powershell/azure-table-storage.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+| Parameter            | Mandatory | Description                                                                                                     |
+| -------------------- | --------- | --------------------------------------------------------------------------------------------------------------- |
+| `ResourceGroupName`  | yes       | The resource group where the Table Storage is located                                                           |
```
Comment (mbraekman) [inline docs/preview/features/powershell/azure-table-storage.md:22]:
```suggestion
| `ResourceGroupName`  | yes       | The resource group where the Storage Account is located                                                           |
```

File: docs/preview/features/powershell/azure-table-storage.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+```powershell
+PS> Create-AzTableStorageAccountTable -ResourceGroupName "stock" -StorageAccountName "admin" -TableName "products"
+# Creating table 'products' in the storage account 'admin'..."
+```
+
+With existing table and re-create:
+
+```powershell
+PS> Create-AzTableStorageAccountTable -ResourceGroupname "stock" -StorageAccountName "admin" -TableName "products"
```
Comment (mbraekman) [inline docs/preview/features/powershell/azure-table-storage.md:39]: Seems like the DeleteAnCreate-flag is missing here.
Is this just missing from the documentation, or is the default behavior of the script to delete and recreate an existing table storage?

File: docs/preview/features/powershell/azure-table-storage.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+| `StorageAccountName` | yes       | The name of the Storage Account to add the table to                                                             |
+| `TableName`          | yes       | The name of the table to add on the Storage Account                                                             |
+| `DeleteAndCreate`    | no        | The optional flag to indicate whether or not a possible already existing table should be deleted and re-created |
+
+**Example**
+
+With non-existing table:
+
+```powershell
+PS> Create-AzTableStorageAccountTable -ResourceGroupName "stock" -StorageAccountName "admin" -TableName "products"
```
Comment (mbraekman) [inline docs/preview/features/powershell/azure-table-storage.md:32]: Perhaps we can rename this to: `Create-AzTableStorage`.
Or do you think that would be to trimmed down?

File: src/Arcus.Scripting.TableStorage/Arcus.Scripting.TableStorage.psm1 [hunk 5]
```diff
@@ -0,0 +1,33 @@
+<#
+ .Synopsis
+  Create a Azure Table Storage based to an Storage Account.
```
Comment (mbraekman) [inline src/Arcus.Scripting.TableStorage/Arcus.Scripting.TableStorage.psm1:3]:
```suggestion
  Create a Table Storage within an Azure Storage Account.
```

File: docs/preview/index.md [hunk 6]
```diff
@@ -21,6 +21,7 @@ For more granular packages we recommend reading the documentation.
 * Automate Azure Data Factory tasks ([powershell](featues/powershell/azure-data-factory))
 * Automate Azure Key Vault tasks ([powershell](features/powershell/azure-key-vault))
 * Automate Azure API Management tasks ([powershell](features/powershell/azure-api-management))
+* Automate Azure Table Storage tasks ([powershell](features/powershell/azure-table-storage))
```
Comment (mbraekman) [inline docs/preview/index.md:24]: Should we create a `azure-table-storage` module, or a `azure-storage` module, which includes operations for table/blob/file/... (all sub-types of an Azure Storage Account)?

I'd tend to go for `azure-storage`, but perhaps that would become to general? 
What do you think: @tomkerkhove?

File: src/Arcus.Scripting.TableStorage/Arcus.Scripting.TableStorage.psm1 [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+ 
+ .Description
+  (Re)Create a Azure Table Storage based to an Storage Account.
```
Comment (mbraekman) [inline src/Arcus.Scripting.TableStorage/Arcus.Scripting.TableStorage.psm1:6]:
```suggestion
  (Re)Create a Table Storage within an Azure Storage Account.
```

File: src/Arcus.Scripting.TableStorage/Arcus.Scripting.TableStorage.psm1 [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+ .Parameter ResourceGroupName
+  The resource group where the Table Storage is located.
```
Comment (mbraekman) [inline src/Arcus.Scripting.TableStorage/Arcus.Scripting.TableStorage.psm1:9]:
```suggestion
  The resource group where the Storage Account is located.
```

File: docs/preview/features/powershell/azure-table-storage.md (hunk 3 above)
Comment (stijnmoreels) [reply docs/preview/features/powershell/azure-table-storage.md:39]: No, it's missing from the docs. The default behavior is not to delete+create it.

File: docs/preview/features/powershell/azure-table-storage.md (hunk 4 above)
Comment (stijnmoreels) [reply docs/preview/features/powershell/azure-table-storage.md:32]: Is good for me. Better even. But I wasn't sure of the 'StorageAccount' that was present in the 'old' script and if it had a meaning it was placed in the name. So I left it there.

**Comment by stijnmoreels** [comment]
> Can you change all `Storage Account` references to include `Azure` as it's a product name please?

Yes, search and renamed.

File: src/Arcus.Scripting.TableStorage/Scripts/Create-AzTableStorageAccountTable.ps1 [hunk 9]
```diff
@@ -0,0 +1,85 @@
+param(
+   [string][parameter(Mandatory = $true)] $ResourceGroup,
+   [string][parameter(Mandatory = $true)] $StorageAccountName,
+   [string][parameter(Mandatory = $true)] $TableName,
+   [switch][parameter()] $DeleteAndCreate = $false
```
Comment (tomkerkhove) [inline src/Arcus.Scripting.TableStorage/Scripts/Create-AzTableStorageAccountTable.ps1:5]: `DeleteAndCreate` or `Recreate`?

Comment (stijnmoreels) [reply src/Arcus.Scripting.TableStorage/Scripts/Create-AzTableStorageAccountTable.ps1]: Choose `Recreate` for this.

File: docs/preview/index.md (hunk 6 above)
Comment (mbraekman) [reply docs/preview/index.md:24]: Agree on `Arcus.Scripting.Storage.Table`
Also agree on adding on overlapping module `Arcus.Scripting.Storage`

File: docs/preview/features/powershell/azure-storage.md [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+---
+
+# Azure Storage
+
+## Installation
+
+To have access to the following features, you have to import the module:
+
+```powershell
+PS> Import-Module -Name Arcus.Scripting.Storage
```
Comment (tomkerkhove) [inline docs/preview/features/powershell/azure-storage.md:13]: Let's use this as a module name
```suggestion
PS> Import-Module -Name Arcus.Scripting.Storage.Table
```

File: src/Arcus.Scripting.Storage/Arcus.Scripting.Storage.psm1 [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+ .Parameter StorageAccountName
+  The name of the Azure Storage Account to add the table to.
+
+ .Parameter TableName
+  The name of the table to add on the Azure Storage Account.
+
+ .Parameter Recreate
+  The optional flag to indicate whether or not a possible already existing table should be deleted and re-created.
+#>
+function Create-AzTableStorage {
```
Comment (mbraekman) [inline src/Arcus.Scripting.Storage/Arcus.Scripting.Storage.psm1:20]: Let's change this to `Create-AzStorageTable` based on input from Tom.

File: src/Arcus.Scripting.Storage/Arcus.Scripting.Storage.psd1 [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+# TypesToProcess = @()
+
+# Format files (.ps1xml) to be loaded when importing this module
+# FormatsToProcess = @()
+
+# Modules to import as nested modules of the module specified in RootModule/ModuleToProcess
+# NestedModules = @()
+
+# Functions to export from this module
+FunctionsToExport = @('Create-AzTableStorage')
```
Comment (mbraekman) [inline src/Arcus.Scripting.Storage/Arcus.Scripting.Storage.psd1:69]: Let's change this to `Create-AzStorageTable` based on input from Tom.

File: docs/preview/features/powershell/azure-storage.md [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+| `StorageAccountName` | yes       | The name of the Azure Storage Account to add the table to                                                             |
+| `TableName`          | yes       | The name of the table to add on the Azure Storage Account                                                             |
+| `Recreate`    | no        | The optional flag to indicate whether or not a possible already existing table should be deleted and re-created |
+
+**Example**
+
+With non-existing table:
+
+```powershell
+PS> Create-AzTableStorageAccountTable -ResourceGroupName "stock" -StorageAccountName "admin" -TableName "products"
```
Comment (mbraekman) [inline docs/preview/features/powershell/azure-storage.md:32]: Update to match the actual command.

File: docs/preview/features/powershell/azure-storage.md [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+# Creating table 'products' in the storage account 'admin'..."
+```
+
+With existing table and re-create:
+
+```powershell
+PS> Create-AzTableStorageAccountTable -ResourceGroupname "stock" -StorageAccountName "admin" -TableName "products" -Recreate
```
Comment (mbraekman) [inline docs/preview/features/powershell/azure-storage.md:39]: Update to match the actual command.

**Pull request merged**